### PR TITLE
Implement user challenge data from GraphQL into the Dapp's Listing and Dashboard pages

### PR DESCRIPTION
**Pull request by am9u**

### CODE REVIEW

File: packages/core/src/utils/listingDataHelpers/challengeHelper.ts [hunk 1]
```diff
@@ -91,7 +92,8 @@ export function isOriginalChallengeVoteOverturned(challengeData: ChallengeData):
   if (challengeData.appeal) {
     if (challengeData.appeal.appealGranted) {
       if (challengeData.appeal.appealChallenge) {
-        if (isVotePassed(challengeData.appeal.appealChallenge.poll)) {
+        const isVotePassed2 = isVotePassed(challengeData.appeal.appealChallenge.poll);
```
Comment (nickreynolds) [inline packages/core/src/utils/listingDataHelpers/challengeHelper.ts]: this might have been me, but I think we can undo this change

File: packages/dapp/src/components/Dashboard/ClaimRewardsItem/ClaimRewardsItemReduxWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+class ClaimRewardsProposalItemReduxWrapperComponent extends React.Component<
+  ClaimRewardsItemOwnProps & ProposalClaimRewardsComponentProps & ClaimRewardsChallengeProp & DispatchProp<any>
+> {
+  public async componentDidMount(): Promise<void> {
+    await this.ensureProposalData();
+  }
+
+  public async componentDidUpdate(): Promise<void> {
+    await this.ensureProposalData();
```
Comment (nickreynolds) [inline packages/dapp/src/components/Dashboard/ClaimRewardsItem/ClaimRewardsItemReduxWrapper.tsx]: should this be only if relevant props have changed?

File: packages/dapp/src/components/Dashboard/MyTasksItem/MyTasksItemReduxComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  makeGetUserAppealChallengeData,
+  getChallengeState,
+  getAppealChallengeState,
+} from "../../../selectors";
+import { MyTasksItemOwnProps, MyTasksItemReduxProps } from "./MyTasksItemTypes";
+import MyTasksItemComponent from "./MyTasksItemComponent";
+
+class MyTasksItemReduxWrapper extends React.Component<MyTasksItemOwnProps & MyTasksItemReduxProps & DispatchProp<any>> {
+  public async componentDidUpdate(): Promise<void> {
+    await this.ensureListingAndNewsroomData();
```
Comment (nickreynolds) [inline packages/dapp/src/components/Dashboard/MyTasksItem/MyTasksItemReduxComponent.tsx]: also should this only be when relevant props have changed?

File: packages/dapp/src/helpers/queryTransformations.ts [hunk 4]
```diff
@@ -207,3 +269,121 @@ export function transformGraphQLDataIntoAppealChallenge(
...
+    } else if (pollData) {
+      canUserReveal = didUserCommit && !didUserReveal && isInRevealStage(pollData);
+    }
+
+    const userChallengeData = {
+      didUserCommit,
+      didUserReveal,
+      didUserCollect,
+      didUserRescue,
+      // didCollectAmount: new BigNumber(didCollectAmount || 0),
```
Comment (nickreynolds) [inline packages/dapp/src/helpers/queryTransformations.ts]: i assume this line should be deleted

File: packages/dapp/src/components/Dashboard/ClaimRewardsItem/ClaimRewardsItemReduxWrapper.tsx (hunk 2 above)
Comment (am9u) [reply packages/dapp/src/components/Dashboard/ClaimRewardsItem/ClaimRewardsItemReduxWrapper.tsx]: `ensureProposalData` does some checking before it dispatches the action to fetch this data.
